### PR TITLE
Add masked language modeling to community examples

### PR DESCRIPTION
**Pull request by yang-zhang**
Add masked language modeling (based on Transformers) to community examples

## What does this PR do?

Adds link to lightning-language-modeling [repo](https://github.com/yang-zhang/lightning-language-modeling) (for fine-tune masked language model with Pytorch Lightning and Huggingface Transformers) to community examples.
I can't find example of using lightning to do language modeling out there so I made one myself. It is based on `Transformers examples/language-modeling` and lightning's examples. I am new to lightning but tried to follow the best practices in lightning's doc. I appreciate feedback to help me better learn to use lightning.

Fixes # (issue)

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In in short, see following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Yes.

### CODE REVIEW

**Comment by SeanNaren** [comment]
This is a neat little example, and very clean! 

Could you by any chance simplify the install to just a `pip install -r requirements.txt`? this will make life easier for install. I think most of those libraries have whls for most python versions

**Comment by yang-zhang** [comment]
> This is a neat little example, and very clean!
> 
> Could you by any chance simplify the install to just a `pip install -r requirements.txt`? this will make life easier for install. I think most of those libraries have whls for most python versions

Thanks. Have done the change.

**Review by SeanNaren — APPROVED** [review]
I really like the repo @yang-zhang! Great job :)

For your validation step here, I'd suggest adding the `sync_dist=True` flag in `self.log`. This will ensure on multiple processes validation is correctly synced across them: 

https://github.com/yang-zhang/lightning-language-modeling/blob/main/language_model.py#L82

cc @tchaton @Borda

**Comment by yang-zhang** [comment]
> I really like the repo @yang-zhang! Great job :)
> 
> For your validation step here, I'd suggest adding the `sync_dist=True` flag in `self.log`. This will ensure on multiple processes validation is correctly synced across them:
> 
> https://github.com/yang-zhang/lightning-language-modeling/blob/main/language_model.py#L82
> 
> cc @tchaton @Borda

Thank you! Updated.

**Comment by rohitgr7** [comment]
Nice!

BTW `LightningDataModule` doesn't have `save_hyperparameters()` yet?
https://github.com/yang-zhang/lightning-language-modeling/blob/15f513cdeb8a448d4bcade3f427257657629f67e/language_model.py#L18-L29

**Comment by edenlightning** [comment]
Great work @yang-zhang! Another improvment i'd suggest is to use DataModule you wrote, instead of the processing in data.py See this example for reference: https://github.com/nateraw/lightning-pretrain-hf/

**Comment by yang-zhang** [comment]
> Nice!
> 
> BTW `LightningDataModule` doesn't have `save_hyperparameters()` yet?
> https://github.com/yang-zhang/lightning-language-modeling/blob/15f513cdeb8a448d4bcade3f427257657629f67e/language_model.py#L18-L29


> Great work @yang-zhang! Another improvment i'd suggest is to use DataModule you wrote, instead of the processing in data.py See this example for reference: https://github.com/nateraw/lightning-pretrain-hf/

Thank you. Do you mean moving the functionalities in data.py into the DataModule?

**Comment by rohitgr7** [comment]
> I tried to find save_hyperparameters() for LightningDataModule but couldn't. Not sure if I missed it or it doesn't exist?

yeah, it doesn't exist yet. It's WIP https://github.com/PyTorchLightning/pytorch-lightning/pull/3792

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Thanks for your work !
